src/pt/animexnovel/src/eu/kanade/tachiyomi/extension/pt/animexnovel/AnimeXNovel.kt

```
@@ -66,19 +66,27 @@ class AnimeXNovel : ZeistManga(
 
         val script = document.select("script")
             .map(Element::data)
-            .firstOrNull(API_KEY_REGEX::containsMatchIn)
+            .firstOrNull(API_KEYS_REGEX::containsMatchIn)
             ?: return emptyList()
 
         val blogId = BLOG_ID_REGEX.find(script)?.groups?.get(1)?.value ?: return emptyList()
-        val apiKey = API_KEY_REGEX.find(script)?.groups?.get(1)?.value ?: return emptyList()
-
-        val url = "https://www.googleapis.com/blogger/v3/blogs/$blogId/posts".toHttpUrl().newBuilder()
-            .addQueryParameter("key", apiKey)
-            .addQueryParameter("labels", label)
-            .addQueryParameter("maxResults", "500")
-            .build()
-
-        val response = client.newCall(GET(url, headers)).execute()
+        val apiKeys = API_KEYS_REGEX.find(script)?.groupValues?.get(1)?.let { content ->
+            API_KEY_REGEX.findAll(content).map { it.groupValues[1] }.toList()
+        }?.takeIf(List<String>::isNotEmpty) ?: emptyList()
```

I think you mean `?: return emptyList()`.

I think the code (including early returns above this) should actually throw the exception on ~L90 instead of returning an empty list. Try extracting a function returning `Response?` and throw the exception when `null` is returned.